lib/serum/build/file_processor.ex

```diff
@@ -113,6 +113,12 @@ defmodule Serum.Build.FileProcessor do
     end
   end
 
+  @doc false
```

Comment:
Duplicate `@doc` and `@spec`.

There are two clauses for `process_posts/2` function, which are `process_posts([], _proj)` and `process_posts(files, proj)`. These two clauses should share the same documentation and function specification.

The following code is my suggestion:

```elixir
@doc false
@spec process_posts([Serum.File.t()], Project.t()) :: Result.t({[Post.t()], [map()]})
# Put a function definition for readability.
def process_posts(files, proj)

# Please use one-liner shorthand if the function body consists of one expression
# and the line fits 80-column boundary.
def process_posts([], _proj), do: {:ok, {[], []}}

def process_posts(files, proj) do
  ...
end
```

Reply:
yeah, i just ran dialyzer:

```
lib/serum/build/file_processor.ex:117: Overloaded contract for 'Elixir.Serum.Build.FileProcessor':process_posts/2 has overlapping domains; such contracts are currently unsupported and are simply ignored
lib/serum/build/file_processor.ex:171: Overloaded contract for 'Elixir.Serum.Build.FileProcessor':generate_lists/2 has overlapping domains; such contracts are currently unsupported and are simply ignored
```

and it wasn't happy with my duplicate `@spec`s.